starknet_devnet/transaction_wrapper.py

```diff
@@ -81,10 +81,11 @@ def __init__(
             "transaction_index": 0 # always the first (and only) tx in the block
         }
 
-        self.trace = {
-            "function_invocation": execution_info.call_info.dump(),
-            "signature": tx_details.to_dict().get("signature", [])
-        }
+        if status is not TxStatus.REJECTED:
```

Comment:
This logic is actually handled in `invoke` and `deploy` in starknet_wrapper.py with the try-except logic. This is also related to my other comment in this second PR review.

Reply:
[3eefb71](https://github.com/Shard-Labs/starknet-devnet/pull/58/commits/3eefb7169025641d5397c45faed6f4278e474405)